File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
from code import interact


def find_armstrong_numbers(numbers):
    finalArray = []
    for i in range(len(numbers)):
        num = list(str(numbers[i]))
        sum = 0
        for j in num:
            sum += pow(int(j), len(num))
        if sum == int(i):
            finalArray.append(sum)
    return finalArray

# print(find_armstrong_numbers(list(range(0,999))))

File: js/armstrongNumbers.js
// How can you make this more scalable and reusable later?

exports.findArmstrongNumbers = function (arr) {
  let list = [];
  for (let i = 0; i < arr.length; i++) {
    let num = String(i).split("");
    let sum = 0;
    for (let i of num) {
      sum += Math.pow(+i, num.length);
    }
    if (sum === i) {
      list.push(sum);
    }
  }
  return list;
};
// console.log(findArmstrongNumbers([]));
